### PR TITLE
Better error message for CUDA error

### PR DESCRIPTION
**Pull request by sklam**
Make new error handling functions to get the CUDA error name and description strings.

### CODE REVIEW

**Comment by nsakharnykh** [comment]
@mtjrider should we mark this as [WIP]?

**Comment by scopatz** [comment]
LGTM other than the merge conflicts.

File: include/gdf/errorutils.h [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 #define CUDA_TRY(x) if ((x)!=cudaSuccess) return GDF_CUDA_ERROR;
 
-#define CUDA_CHECK_LAST() CUDA_TRY(cudaGetLastError())
+#define CUDA_CHECK_LAST() CUDA_TRY(cudaPeekAtLastError())
```
[the remark this answers is not in the document]
Comment (sklam) [reply include/gdf/errorutils.h]: No, `cudaGetLastError` would clear the error. `cudaPeekAtLastError` would not.  This change is necessary to get the error string later.

**Comment by mike-wendt** [comment]
run tests

**Comment by kkraus14** [comment]
Closing and reopening to trigger CI.

**Comment by kkraus14** [comment]
@sklam I fixed the merge conflict, is this ready for review?

**Pull request merged**